clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java

```diff
@@ -17,90 +17,384 @@
+          convertedUnit = MemoryUnit.PB;
+          break;
+        default:
+          throw new UnsupportedOperationException("Unrecognized unit " + unit);
+      }
+      return convertedUnit;
+    }
+
+    private PassthroughServerBuilder resource(String resourceName, int size, MemoryUnit unit) {
+      final ResourceType resource = new ResourceType();
+      resource.setName(resourceName);
+      resource.setUnit(unit);
+      resource.setValue(BigInteger.valueOf((long)size));
+      this.resources.getResource().add(resource);
+      return this;
+    }
+
+    public PassthroughServerBuilder serviceProvider(ServiceProvider serviceProvider, ServiceProviderConfiguration configuration) {
+      this.serviceProviders.put(serviceProvider, configuration);
+      return this;
+    }
+
+    public PassthroughServerBuilder serverEntityService(ServerEntityService<?, ?> service) {
+      this.serverEntityServices.add(service);
+      return this;
+    }
+
+    public PassthroughServerBuilder clientEntityService(EntityClientService<?, ?, ? extends EntityMessage, ? extends EntityResponse> service) {
+      this.clientEntityServices.add(service);
+      return this;
+    }
+
+    public PassthroughServer build() {
+      PassthroughServer newServer = new PassthroughServer(true);
+
+      newServer.registerServerEntityService(new EhcacheServerEntityService());
+      newServer.registerClientEntityService(new EhcacheClientEntityService());
+      newServer.registerServerEntityService(new CoordinationServerEntityService());
+      newServer.registerClientEntityService(new ClientCoordinationEntityService());
+
+      for (ServerEntityService<?, ?> service : serverEntityServices) {
+        newServer.registerServerEntityService(service);
+      }
+
+      for (EntityClientService<?, ?, ? extends EntityMessage, ? extends EntityResponse> service : clientEntityServices) {
+        newServer.registerClientEntityService(service);
+      }
+
+      if (!this.resources.getResource().isEmpty()) {
+        newServer.registerServiceProvider(new OffHeapResourcesProvider(), new OffHeapResourcesConfiguration(this.resources));
+      }
+
+      for (Map.Entry<ServiceProvider, ServiceProviderConfiguration> entry : serviceProviders.entrySet()) {
+        newServer.registerServiceProvider(entry.getKey(), entry.getValue());
+      }
+
+      return newServer;
+    }
   }
 
-  public static synchronized PassthroughServer server() {
-    return server;
+  public static Collection<Properties> getConnectionProperties(URI uri) {
+    ServerDescriptor serverDescriptor = servers.get(createKey(uri));
+    if (serverDescriptor != null) {
+      return serverDescriptor.getConnections().values();
+    } else {
+      return Collections.emptyList();
+    }
   }
 
   @Override
   public boolean handlesURI(URI uri) {
-    String hostName = uri.getHost();
-    return hostName.endsWith("example.com") || hostName.equals("localhost");
+    return servers.containsKey(createKey(uri));
```

Comment:
The servers are registered using the result of `createKey` ... the lookup needs to be that way as well.